dali/pipeline/operators/decoder/decoder_cache_blob_test.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <memory>
+#include "dali/pipeline/operators/decoder/decoder_cache_blob.h"
+
+namespace dali {
+namespace testing {
+
+const char kKey1[] = "file1.jpg";
+const std::vector<uint8_t> kValue1(300, 0xAA);
+const Dims kDims1{100, 1, 3};
+
+struct DecoderCacheBlobTest : public ::testing::Test {
+  DecoderCacheBlobTest() {
+  }
+
+  void SetUp() override {
+    SetUpImpl((1<<9));
+  }
+
+  void SetUpImpl(std::size_t cache_size, std::size_t image_size_threshold = 0) {
+    cache_.reset(new DecoderCacheBlob(cache_size, image_size_threshold));
+  }
+
+  std::unique_ptr<DecoderCacheBlob> cache_;
+};
+
+TEST_F(DecoderCacheBlobTest, EmptyCache) {
+  EXPECT_FALSE(cache_->IsCached(kKey1));
+}
+
+TEST_F(DecoderCacheBlobTest, Add) {
+  EXPECT_FALSE(cache_->IsCached(kKey1));
+  cache_->Add(kKey1, &kValue1[0], kValue1.size(), kDims1);
+  EXPECT_TRUE(cache_->IsCached(kKey1));
+  std::vector<uint8_t> cachedData(kValue1.size(), 0x00);
```

Comment:
I think initialization with `kValue1.size()` is superfluous.

Reply:
why? I am allocating a big enough vector, that's all

Reply:
Did you mean that I should do `std::vector<uint8_t> cachedData(kValue1.size());` instead?